torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -52,63 +52,60 @@ bool isBroadcasting(Node* node) {
   return broadcasting.count(node->kind());
 }
 
-// First iterate over the 'from' tensor sizes. Ignore all leading and trailing
-// dimensions that are simply one, since they can be trivially broadcasted.
-// When iterating over the dimension sizes (with reduced 'from' tensor),
-// starting at the trailing dimension, the dimension sizes must either be equal,
-// or one of them does not exist. If a broadcast candidate is not found at the
-// trailing dimension, search at the leading dimension. If one is found here,
-// return the `axis` argument to be emitted to ONNX on the broadcasting operator
+// This boils down to
+//
+//     Treating leading and trailing `1`s in `from` as wildcards, is
+//     `from` a substring of `to`, and if so starting at which index?
+//
+// To answer it we do a naive, intentionally n-squared substring
+// search, tweaked to have the correct wildcard behavior.
```

Comment:
Can you explain in this comment why greedy match doesn't work, with an example?

Reply:
what exactly do you mean by greedy match? The old behavior, where it had to be either a prefix or a suffix?